src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Indentation/CSharpIndentationService.Indenter.cs

```diff
@@ -84,9 +84,30 @@ private static IndentationResult GetIndentationBasedOnToken(Indenter indenter, S
             if (token.IsKind(SyntaxKind.MultiLineRawStringLiteralToken))
             {
                 var endLine = sourceText.Lines.GetLineFromPosition(token.Span.End);
-                var nonWhitespaceOffset = endLine.GetFirstNonWhitespaceOffset();
-                Contract.ThrowIfNull(nonWhitespaceOffset);
-                return new IndentationResult(indenter.LineToBeIndented.Start, nonWhitespaceOffset.Value);
+                var minimumOffset = endLine.GetFirstNonWhitespaceOffset();
+                Contract.ThrowIfNull(minimumOffset);
+
+                // If possible, indent to match the indentation of the previous non-whitespace line contained in the
+                // same raw string. Otherwise, indent to match the ending line of the raw string.
+                var startLine = sourceText.Lines.GetLineFromPosition(token.SpanStart);
+                for (var currentLineNumber = indenter.LineToBeIndented.LineNumber - 1; currentLineNumber >= startLine.LineNumber + 1; currentLineNumber--)
```

Comment:
what if currentLineNumber ends up either being negative, or gets the line the raw token starts on?  should we ensure that we start at Max(LineNumber - 1, startLine + 1) ?

Reply:
It stops at `startLine.LineNumber + 1`, case is covered by `RawString1`.

Reply:
can you also add tests that have the raw string on line 0 (which would be legal in top level statements).